msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/PublicClientApplication.java

```diff
@@ -66,10 +66,9 @@ public CompletableFuture<IAuthenticationResult> acquireToken(IntegratedWindowsAu
     @Override
     public CompletableFuture<IAuthenticationResult> acquireToken(DeviceCodeFlowParameters parameters) {
 
-        if (!(AuthorityType.AAD.equals(authenticationAuthority.authorityType()) ||
-                AuthorityType.ADFS.equals(authenticationAuthority.authorityType()))) {
+        if (AuthorityType.B2C.equals(authenticationAuthority.authorityType())) {
             throw new IllegalArgumentException(
-                    "Invalid authority type. Device Flow is only supported by AAD and ADFS authorities");
+                    "Invalid authority type. Device Flow is only supported by AAD, ADFS and CIAM authorities");
```

Comment:
Since you already changed the `if` clause above, you can also change this error message to "... Device Flow is not supported by B2C...". That way, this function becomes future-proof when/if we will support a new type of authority in the future that also supports Device Flow.